.github/workflows/packages.yml

```diff
@@ -251,69 +206,56 @@ jobs:
 
         # Publishing repository changes.
         if [ "$uploaded_files" = "true" ]; then
-          echo "[$(date +%H:%M:%S)] Adding packages to repository '$REPOSITORY_NAME'..."
-          http_status_code=""
-          curl_response=$(
-            curl \
-              --silent \
-              --retry 2 \
-              --retry-delay 3 \
-              --max-time 300 \
-              --user "${{ secrets.APTLY_API_AUTH }}" \
-              --user-agent "Termux-Packages/1.0 (https://github.com/termux/termux-packages)" \
-              --request POST \
-              --write-out "|%{http_code}" \
-              https://packages-cf.termux.org/aptly-api/repos/${REPOSITORY_NAME}/file/${REPOSITORY_NAME}-${{ github.sha }} || true
-          )
-          http_status_code=$(echo "$curl_response" | cut -d'|' -f2 | grep -oP '\d{3}$')
-
-          if [ "$http_status_code" = "200" ]; then
-            warnings=$(echo "$curl_response" | cut -d'|' -f1 | jq '.Report.Warnings' | jq -r '.[]')
-            if [ -n "$warnings" ]; then
-              echo "[$(date +%H:%M:%S)] APTLY WARNINGS (NON-CRITICAL):"
-              echo
-              echo "$warnings"
-              echo
-            fi
-          else
-            echo "[$(date +%H:%M:%S)] Error: got http_status_code == '$http_status_code', packages may not appear in repository."
-          fi
+          aptly_add_to_repo
 
           # Usually temporary directory is deleted automatically, but in certain cases it is left.
           aptly_delete_dir
 
           # Final part to make changes appear in web root.
           echo "[$(date +%H:%M:%S)] Publishing repository changes..."
-          http_status_code=""
-          curl_response=$(
-            curl \
-              --silent \
-              --retry 2 \
-              --retry-delay 3 \
-              --max-time 300 \
-              --user "${{ secrets.APTLY_API_AUTH }}" \
-              --user-agent "Termux-Packages/1.0 (https://github.com/termux/termux-packages)" \
-              --header 'Content-Type: application/json' \
-              --request PUT \
-              --data '{"Signing": {"Passphrase": "${{ secrets.GPG_PASSPHRASE }}"}}' \
-              --write-out '|%{http_code}' \
-              https://packages-cf.termux.org/aptly-api/publish/${REPOSITORY_NAME}/${REPOSITORY_DISTRIBUTION} || true
-          )
-          http_status_code=$(echo "$curl_response" | cut -d'|' -f2 | grep -oP '\d{3}$')
+          if ! aptly_publish_repo; then
+            exit 1
+          fi
+        fi
+    - name: Upload to grimler.se
+      # Run even if upload to packages.termux.org failed:
+      if: always()
```

Comment:
Better read this from GitHub secrets instead of forcing it to always on.

Here's why:-
- There are times when unstable updates are pushed to main repo to rebuild dependent packages against latest version. (Take for example OpenSSL upgrades and other stuff)
- (Some other reason?)

Reply:
You mean so that we can pause updates to this repo at times? That will still be possible since the changes [aren't published by CI](https://github.com/termux/termux-packages/blob/cf632b5fe597bcc094c5968880b13f0d4f52878e/.github/workflows/packages.yml#L256), so I still control when/if updates appear on the repo